src/components/CustomDropdownComp.tsx

```diff
@@ -33,161 +33,157 @@ export interface CustomDropdownComponentProps {
-            return createOption(firstLabel, secondLabel, imgUrl);
-          })
-          setOptions(options);
+        if (props.options.status === 'available') {
+            const options = props.options.items.map(obj => {
+                let { firstLabel, secondLabel, imgUrl }: { firstLabel: string; secondLabel: string; imgUrl: string; } = getLabelValues(obj);
+                return createOption(firstLabel, secondLabel, imgUrl);
+            })
+            setOptions(options);
         }
-      }, [props.options]);
+    }, [props.options]);
 
-      const handleChange = async (inputValue: any, actionMeta: any) => {
-        if (
-            actionMeta.action === 'select-option'
-        ) {
-          setIsLoading(true);
-          try {
-            if(props.contextObjLabel.status==='available'){
-              props.contextObjLabel.setValue(inputValue.value);
-            }
-            if(props.selectOption.canExecute){
-              props.selectOption.execute();
+    const handleChange = async (inputValue: any, actionMeta: any) => {
+        if (actionMeta.action === 'select-option') {
+            setIsLoading(true);
+            try {
+                if (props.contextObjLabel.status === 'available') {
+                    props.contextObjLabel.setValue(inputValue.value);
+                }
+                if (props.selectOption.canExecute) {
+                    props.selectOption.execute();
+                }
+                setValue(createOption(inputValue.value, inputValue.secondLabel, inputValue.url));
+            } catch (err) {
+                console.error('Failed to select a Tag: ' + err);
             }
-            setValue(createOption(inputValue.value, inputValue.secondLabel , inputValue.url));
-          } catch (err) {
-            console.error('Failed to select a Tag: ' + err);
-          }
-          setIsLoading(false);
+            setIsLoading(false);
         }
-        if (
-          actionMeta.action === 'create-option'
-        ) {
-          setIsLoading(true);
-          try {
-            if(props.contextObjLabel.status==='available'){
-              props.contextObjLabel.setValue(inputValue.value);
-            }
-            if(props.createValue.canExecute){
-              props.createValue.execute();
+
+        if (actionMeta.action === 'create-option') {
+            setIsLoading(true);
+            try {
+                if (props.contextObjLabel.status === 'available') {
+                    props.contextObjLabel.setValue(inputValue.value);
+                }
+                if (props.createValue.canExecute) {
+                    props.createValue.execute();
+                }
+                if (!props.enableCreate) {
+                    setValue(createOption(inputValue.value, '', ''));
+                    setOptions([...options, createOption(inputValue.value, '', '')])
+                }
+            } catch (err) {
+                console.error('Failed to create a Tag: ' + err);
```

Comment:
Is there no more graceful way to handle this case than to log an error on the console?